frontend/js/src/search/SongSearch.tsx

```diff
@@ -0,0 +1,166 @@
+import { getObjectForURLSearchParams } from "../utils/utils";
+import Pagination from "../common/Pagination";
+
+const RECORDING_COUNT_PER_PAGE = 50;
+
+type SongSearchProps = {
+  searchQuery: string;
+};
+
+type SongTypeSearchResult = {
+  count: number;
+  offset: number;
+  recordings: {
+    id: string;
+    title: string;
+    "artist-credit": {
+      name: string;
+      joinphrase?: string;
+      artist: {
+        id: string;
+        name: string;
+      };
+    }[];
+    releases: {
+      id: string;
+      title: string;
+      "artist-credit": {
+        name: string;
+        joinphrase?: string;
+      }[];
+      "release-groups": {
+        id: string;
+        title: string;
+      }[];
+    }[];
+  }[];
+};
+
+export default function SongSearch(props: SongSearchProps) {
+  const { APIService } = React.useContext(GlobalAppContext);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const searchParamsObj = getObjectForURLSearchParams(searchParams);
+  const currPageNoStr = searchParams.get("page") || "1";
+  const currPageNo = parseInt(currPageNoStr, 10);
+
+  const { searchQuery } = props;
+
+  const { data: loaderData, isLoading: loading } = useQuery({
+    queryKey: ["search-song", searchQuery, currPageNoStr],
+    queryFn: async () => {
+      try {
+        const offset = (currPageNo - 1) * RECORDING_COUNT_PER_PAGE;
+        const queryData = await APIService.recordingLookup(
+          searchQuery,
+          offset,
+          RECORDING_COUNT_PER_PAGE
+        );
+        return {
+          data: queryData as SongTypeSearchResult,
+          hasError: false,
+          errorMessage: "",
+        };
+      } catch (error) {
+        return {
+          data: {} as SongTypeSearchResult,
+          hasError: true,
+          errorMessage: error.message,
+        };
+      }
+    },
+  });
+
+  const {
+    data: rawData = {} as SongTypeSearchResult,
+    hasError = false,
+    errorMessage = "",
+  } = loaderData || {};
+
+  const { recordings = [] } = rawData;
+  const totalPageCount = Math.ceil(rawData.count / RECORDING_COUNT_PER_PAGE);
+
+  const handleClickPrevious = () => {
+    setSearchParams({
+      ...searchParamsObj,
+      page: Math.max(currPageNo - 1, 1).toString(),
+    });
+  };
+
+  const handleClickNext = () => {
+    setSearchParams({
+      ...searchParamsObj,
+      page: Math.min(currPageNo + 1, totalPageCount).toString(),
+    });
+  };
+
+  const getListenCard = (recording: SongTypeSearchResult["recordings"][0]) => {
+    const artists: MBIDMappingArtist[] = [];
+    const artistMBIDs: Array<string> = [];
+    const artistCredit = recording["artist-credit"];
+    artistCredit.map((ac) => {
+      artists.push({
+        artist_mbid: ac.artist.id,
+        artist_credit_name: ac.name,
+        join_phrase: ac.joinphrase ?? "",
+      });
+      artistMBIDs.push(ac.artist.id);
+      return null;
+    });
+
+    return (
+      <ListenCard
+        key={recording.id}
+        listen={{
```

Comment:
For readability, perhaps best to defined the listen above like you did for artistCredit